### PR TITLE
Update supported versions

### PR DESCRIPTION
**Pull request by hugovk**
Add:

* Python 3.7-3.9
* Python 3.10: currently in RC

Remove:

* Python 2.7, 3.4-3.5: EOL
* ~Python 3.6: to match Matplotlib/[NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html)~

Run:

* Upgrade Python syntax with `pyupgrade --py36-plus`

### CODE REVIEW

**Review by timhoffm — COMMENTED** [review]
I suggest to still support 3.6. AFAICS there is no technical reason to not support it.

**Comment by hugovk** [comment]
Sure, reverted!

File: cycler.py [hunk 1]
```diff
@@ -263,7 +257,7 @@ def __add__(self, other):
         """
         if len(self) != len(other):
             raise ValueError("Can only add equal length cycles, "
-                             "not {0} and {1}".format(len(self), len(other)))
+                             "not {} and {}".format(len(self), len(other)))
```
Comment (timhoffm) [inline cycler.py:260]: f-stringify?

**Review by timhoffm — APPROVED** [review]

File: cycler.py (hunk 1 above)
Comment (hugovk) [reply cycler.py]: Updated!

**Review by QuLogic — APPROVED** [review]

**Pull request merged**